x/interchainquery/keeper/msg_submit_query_response_test.go

```diff
@@ -91,33 +94,64 @@ func (s *KeeperTestSuite) TestMsgSubmitQueryResponse_UnknownId() {
 	s.Require().True(found)
 }
 
-func (s *KeeperTestSuite) TestMsgSubmitQueryResponse_ExceededTtl() {
+func (s *KeeperTestSuite) TestMsgSubmitQueryResponse_Timeout_RejectQuery() {
 	tc := s.SetupMsgSubmitQueryResponse()
 
-	// Remove key from the query type so to bypass the VerifyKeyProof function
-	tc.query.QueryType = strings.ReplaceAll(tc.query.QueryType, "key", "")
+	// set timeout to be expired and set the policy to reject
+	tc.query.TimeoutTimestamp = uint64(1)
+	tc.query.TimeoutPolicy = types.TimeoutPolicy_REJECT_QUERY_RESPONSE
+	s.App.InterchainqueryKeeper.SetQuery(s.Ctx, tc.query)
+
+	_, err := s.GetMsgServer().SubmitQueryResponse(tc.goCtx, &tc.validMsg)
+	s.Require().NoError(err)
 
-	// set timeout to be expired
+	// check that the query original query was deleted,
+	_, found := s.App.InterchainqueryKeeper.GetQuery(s.Ctx, tc.query.Id)
+	s.Require().False(found, "original query should be removed")
+}
+
+func (s *KeeperTestSuite) TestMsgSubmitQueryResponse_Timeout_RetryQuery() {
+	tc := s.SetupMsgSubmitQueryResponse()
+
+	// set timeout to be expired and set the policy to retry
 	tc.query.TimeoutTimestamp = uint64(1)
+	tc.query.TimeoutPolicy = types.TimeoutPolicy_RETRY_QUERY_REQUEST
 	s.App.InterchainqueryKeeper.SetQuery(s.Ctx, tc.query)
 
-	resp, err := s.GetMsgServer().SubmitQueryResponse(tc.goCtx, &tc.validMsg)
+	_, err := s.GetMsgServer().SubmitQueryResponse(tc.goCtx, &tc.validMsg)
 	s.Require().NoError(err)
-	s.Require().NotNil(resp)
 
-	// check that the query was not deleted (since the query timed out)
+	// check that the query original query was deleted,
+	// but that a new one was created for the retry
 	_, found := s.App.InterchainqueryKeeper.GetQuery(s.Ctx, tc.query.Id)
-	s.Require().True(found)
+	s.Require().False(found, "original query should be removed")
+	s.Require().Len(s.App.InterchainqueryKeeper.AllQueries(s.Ctx), 1, "there should be one new query")
 }
 
-func (s *KeeperTestSuite) TestMsgSubmitQueryResponse_FindAndInvokeCallback_WrongHostZone() {
+func (s *KeeperTestSuite) TestMsgSubmitQueryResponse_Timeout_ExecuteCallback() {
 	tc := s.SetupMsgSubmitQueryResponse()
 
+	// set timeout to be expired and set the policy to retry
+	tc.query.TimeoutTimestamp = uint64(1)
+	tc.query.TimeoutPolicy = types.TimeoutPolicy_EXECUTE_QUERY_CALLBACK
 	s.App.InterchainqueryKeeper.SetQuery(s.Ctx, tc.query)
 
 	// rather than testing by executing the callback in its entirety,
 	//   check by invoking it without a registered host zone and catching the appropriate error
```

Comment:
just out of curiosity, why test it via this failure path? if possible to test the success case, I think we should! possible the proof checking logic will break the unit test...

Reply:
this test belongs to the ICQ module and is just meant to test the generic case of timeout. So I don't think it makes sense to bundle the logic from one of our stakeibc callbacks here (we'd essentially be duplicating the other queries unit test and would have to mock out a ton of state)

That said, I don't love the that this just checks for a random chainId error so if you have a suggestion on how to clean that up, i'm all ears!

Reply:
It feels less robust than it should be, but I don't have a concrete proposal for a solution -- I can't think of a way to bolster the test case. Maybe we leave it for now unless @ethan-stride has any ideas.

Reply:
leaving as is